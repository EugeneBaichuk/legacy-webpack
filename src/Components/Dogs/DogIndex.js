import React, { useState, useEffect } from 'react';
import { Button, Spinner } from 'reactstrap';
import axios from 'axios';

import './Dogs.css';

const DogIndex = ({text}) => {
  const [url, setUrl] = useState('');

  const handleFetchDogImg = async () => {
    const res = await axios.get('https://dog.ceo/api/breeds/image/random');
    setUrl(res.data.message);
  };

  useEffect(() => {
    handleFetchDogImg();
  }, []);

  return (
    <div className="content">
      <div className="container">
        <section className="section">
          <div className="d-flex justify-content-center mb-4">
            <Button color="secondary" onClick={handleFetchDogImg}>
              {text}
            </Button>
          </div>
          <div className="d-flex justify-content-center align-items-center border dog-container">
            {url ? 
              <img src={url} alt="Dog" className='dog-img' /> :
              <Spinner />
            }
          </div>
        </section>
      </div>
    </div>
  );
};

export default DogIndex;