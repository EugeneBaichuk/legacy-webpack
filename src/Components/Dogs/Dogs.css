.dog-container {
  width: 400px; 
  height: 400px; 
  margin: auto; 
  border-radius: 8px;
}
  
.dog-img {
  max-width: 100%; 
  max-height: 100%; 
  object-fit: cover; 
  border-radius: 8px;
}