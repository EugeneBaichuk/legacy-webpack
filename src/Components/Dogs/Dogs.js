import React from 'react';
import DogIndex from './DogIndex';
import Layout from '../Common/Layout';
import { dogsHeader, dogsTaskList } from '../../Config/dogs.constants';
import { dogBtnText } from '../../Config/dogs.constants';

export const Dogs = () => (
  <Layout 
    header={dogsHeader} 
    list={dogsTaskList}  
  >
    <DogIndex text={dogBtnText}/>
  </Layout>
);

