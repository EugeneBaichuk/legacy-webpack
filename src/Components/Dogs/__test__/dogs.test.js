import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Dogs } from '../Dogs';

describe('Cats component', () => {
  afterEach(cleanup);

  test('renders the header correctly', () => {
    const { asFragment } = render(<Dogs />);
    expect(asFragment(<Dogs />)).toMatchSnapshot();
  });
});