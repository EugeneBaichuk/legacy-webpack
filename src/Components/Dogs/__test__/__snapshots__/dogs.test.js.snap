// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cats component renders the header correctly 1`] = `
<DocumentFragment>
  <div
    class="layout container"
  >
    <h1
      class="mb-4 mt-4"
    >
      For this challenge, write the logic to accomplish the following:
    </h1>
    <ul
      class="list-group"
    >
      <li
        class="list-group-item"
      >
        Fetch a random image from this api: 
        <a
          href="https://dog.ceo/dog-api/"
        >
          here
        </a>
      </li>
      <li
        class="list-group-item"
      >
        Save this image URL to the state of the DogIndex component
      </li>
      <li
        class="list-group-item"
      >
        Render the image to the screen
      </li>
      <li
        class="list-group-item"
      >
        Have a button that fetches a new image
      </li>
      <li
        class="list-group-item"
      >
        Make sure to include some good practice error handling
      </li>
    </ul>
    <hr
      class="my-4"
    />
    <div
      class="content"
    >
      <div
        class="container"
      >
        <section
          class="section"
        >
          <div
            class="d-flex justify-content-center mb-4"
          >
            <button
              class="btn btn-secondary"
              type="button"
            >
              Dog eat Dog!
            </button>
          </div>
          <div
            class="d-flex justify-content-center align-items-center border dog-container"
          >
            <div
              class="spinner-border"
              role="status"
            >
              <span
                class="visually-hidden"
              >
                Loading...
              </span>
            </div>
          </div>
        </section>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
